Makefile

```diff
@@ -103,8 +110,16 @@ typedoc:
 docs: typedoc generated-documentation
 
 .PHONY: view-docs
-view-docs: typedoc
-	open ${ROOTDIR}/docs/index.html
+view-docs: docs
+	PYTHONPATH=${ROOTDIR} PIPENV_IGNORE_VIRTUALENVS=1 expect -c 'spawn pipenv run mkdocs serve; expect "Serving on"; exec open "http://localhost:8000"; interact'
+
+.PHONY: publish-docs
+publish-docs:
+	if [ -z ${shell git status -uno | grep "Your branch is up to date with 'origin/main'"} ]; then \
+		echo "The documentation can only be published from main at head."; \
+		exit 1; \
+	fi
+	${PIPENV} run mkdocs gh-deploy
```

Comment:
What does this actually do?

Reply:
The description is here:

https://www.mkdocs.org/user-guide/deploying-your-docs/#github-pages

When I tested it in my fork it worked perfectly, pushing the built docs into the `gh-pages` branch.